src/PhpBrew/VariantBuilder.php

```diff
@@ -767,6 +767,24 @@ public function __construct()
             return $parameters->withOption('--with-pear', $value);
         };
 
+        /*
+         * --with-snmp option
+         *
+         * --with-snmp=[dir]
+         *
+         * On ubuntu you need to install libsnmp-dev
+         */
+        $this->variants['snmp'] = function (ConfigureParameters $parameters, Build $build, $value) {
+            $prefix = Utils::findPrefix(array(
+                new UserProvidedPrefix($value),
+                new BrewPrefixFinder('net-snmp'),
+                new PkgConfigPrefixFinder('netsnmp'),
+                new IncludePrefixFinder('net-snmp/net-snmp-config.h'),
```

Comment:
In the macOS, it can use the `brew --prefix` after using the `brew install net-snmp` command.

```bash
cloudxmacs-Mac-3:~ cloudxmac$ brew --prefix net-snmp
/usr/local/opt/net-snmp
cloudxmacs-Mac-3:~ cloudxmac$ 
cloudxmacs-Mac-3:~ cloudxmac$ sw_vers
ProductName:	macOS
ProductVersion:	11.6
BuildVersion:	20G165
```

Since the Ubuntu 20.04 is released, the `libsnmp-dev` `5.8` version is included.

In the Ubuntu 20.04+, it can use the following command to find the `netsnmp` configuration path:

```
pkg-config --variable=prefix netsnmp
```

In the Ubuntu 18.04+, it can also use the `net-snmp/net-snmp-config.h` file path to find the current `net-snmp` prefix.

Reply:
add your comment into the code directly?

Reply:
Sure. Letting me think how to add these comments properly.

Once adding these comments, I think it's ready to merge the PR.

Reply:
These related comments are added and it's ready to merge the PR now :).